server/resp/src/test/java/org/infinispan/server/resp/JsonCommandsTest.java

```diff
@@ -545,7 +545,40 @@ public void testJSONSETWhiteSpaces() {
       assertThat(compareJSONGet(result, jv)).isTrue();
    }
 
-   // Lattuce Json object doesn't implement comparison. Implementing here
+   @Test
+   public void testJSONSTRLEN() {
+      JsonPath jpDollar = new JsonPath("$");
+      JsonValue jv = defaultJsonParser.createJsonValue("""
+               {"a":"foo", "nested": {"a": "hello"}, "nested2": {"a": 31}}
+            """);
+
+      String key = "doc";
+      // JSON.SET doc $ '{"a":"foo", "nested": {"a": "hello"}, "nested2": {"a": 31}}'
+      assertThat(redis.jsonSet(key, jpDollar, jv)).isEqualTo("OK");
+
+      //  JSON.STRLEN notExist $
+      assertThatThrownBy(() -> {
+         redis.jsonStrlen("notExist", jpDollar);
+      }).isInstanceOf(RedisCommandExecutionException.class)
+              .hasMessage("ERR could not perform this operation on a key that doesn't exist");
+
+      //  JSON.STRLEN doc $
+      assertThat(redis.jsonStrlen(key, jpDollar)).hasSize(1);
+      assertThat(redis.jsonStrlen(key, jpDollar).get(0)).isNull();
+
+      //  JSON.STRLEN doc $..a
+      assertThat(redis.jsonStrlen(key, new JsonPath("$..a"))).containsExactly(3L, 5L, null);
+
+      redis.set("errorRaise", "world");
+
+      //  JSON.STRLEN errorRaise $
+      assertThatThrownBy(() -> {
+         redis.jsonStrlen("errorRaise", jpDollar);
+      }).isInstanceOf(RedisCommandExecutionException.class)
+              .hasMessage("ERR Existing key has wrong Redis type");
+   }
+
```

Comment:
test with `legacy path` are needed. See the note in [json.get](https://redis.io/docs/latest/commands/json.get/) on how commands have different behaviours with legacy path.
I.e.:
```
127.0.0.1:6379> json.get k1
"{\"k1\":\"v1\",\"k2\":\"v2\"}"
127.0.0.1:6379> json.strlen k1   # no path using legacy
(error) WRONGTYPE wrong type of path value - expected string but found object
127.0.0.1:6379> json.strlen k1 $ # jsonpath
1) (nil)
127.0.0.1:6379> json.strlen k1 $.k1 #jsonpath
1) (integer) 2   # return array
127.0.0.1:6379> json.strlen k1 .k1 #legacy
(integer) 2       #return integer
127.0.0.1:6379> 
```